fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/JunoOnboardingMapper.kt

```diff
@@ -13,23 +18,80 @@ import org.mozilla.fenix.settings.SupportUtils
  * Returns a list of all the required Nimbus 'cards' that have been converted to [OnboardingPageUiData].
  */
 internal fun Collection<OnboardingCardData>.toPageUiData(
+    context: Context,
     showNotificationPage: Boolean,
     showAddWidgetPage: Boolean,
-): List<OnboardingPageUiData> =
-    filter {
-        when (it.cardType) {
-            OnboardingCardType.NOTIFICATION_PERMISSION -> {
-                it.enabled && showNotificationPage
-            }
-            OnboardingCardType.ADD_SEARCH_WIDGET -> {
-                it.enabled && showAddWidgetPage
-            }
-            else -> {
-                it.enabled
+): List<OnboardingPageUiData> {
+    val junoOnboardingFeature = FxNimbus.features.junoOnboarding.value()
+    val jexlConditions = junoOnboardingFeature.conditions
+
+    return filter { it.shouldDisplayCard(context, jexlConditions) }
+        .filter {
+            when (it.cardType) {
+                OnboardingCardType.NOTIFICATION_PERMISSION -> {
+                    it.enabled && showNotificationPage
+                }
+
+                OnboardingCardType.ADD_SEARCH_WIDGET -> {
+                    it.enabled && showAddWidgetPage
+                }
+
+                else -> {
+                    it.enabled
+                }
             }
-        }
-    }.sortedBy { it.ordering }
+        }.sortedBy { it.ordering }
         .map { it.toPageUiData() }
+}
+
+private fun OnboardingCardData.shouldDisplayCard(
+    context: Context,
+    jexlConditions: Map<String, String>,
+): Boolean {
+    val jexlCache: MutableMap<String, Boolean> = mutableMapOf()
+    val jexlHelper = context.components.analytics.messagingStorage.helper
+
+    return verifyConditionEligibility(
+        prerequisites,
+        jexlConditions,
+        jexlCache,
+        jexlHelper,
+    ) && !verifyConditionEligibility(disqualifiers, jexlConditions, jexlCache, jexlHelper)
+}
+
+private fun verifyConditionEligibility(
```

Comment:
can we add some tests that cover all branches in this function? Should be able to do something similar to `NimbusMessagingStorageTest`. I think at the moment it's only testing against the 'happy' path?